src/FakeItEasy.Analyzer.Shared/SymbolExtensions.cs

```diff
@@ -29,7 +29,7 @@ public static string GetFullName(this INamedTypeSymbol type)
             }
 
             var containingNamespace = type.ContainingNamespace;
-            while (!string.IsNullOrEmpty(containingNamespace?.Name))
+            while (containingNamespace is object && !string.IsNullOrEmpty(containingNamespace.Name))

```

Comment:
I'm not sure I see the point in this change... What is gained here?

Reply:
Oh. That's weird... `IsNullOrEmpty` should be properly decorated to indicate that the value can't be null when it returns false. Or maybe it is, but only for netcoreapp3.0 / netstandard2.1...
I have a feeling this kind of caveats are going to be *very* annoying in the near future...

Reply:
It might be. I didn't check, but the warning showed for 5 TFMs.

It might not be just `IsNullOrEmpty`'s fault/problem, but the combination with the `?.`, since the warning showed on line 34. The compiler might know that `containingNamespace.Name` isn't null, but may not realize there that reaching line 34 means that `containingNamespace` isn't null. It sounds silly when I say it out loud, but still.